Kafdrop 4.0.0 Docker image doesn't start

Hello!

Today we restarted one of our Docker Swarm system and the Kafdrop service refused to start up properly. The reason was quickly revealed: the :latest image we used in our compose file was updated so it contains Kafdrop version 4.0.0 now. Here is the service log: 
```
:::    :::     :::     :::::::::: :::::::::  :::::::::   ::::::::  :::::::::           :::    
:+:   :+:    :+: :+:   :+:        :+:    :+: :+:    :+: :+:    :+: :+:    :+:         :+:     
+:+  +:+    +:+   +:+  +:+        +:+    +:+ +:+    +:+ +:+    +:+ +:+    +:+        +:+ +:+  
+#++:++    +#++:++#++: :#::+::#   +#+    +:+ +#++:++#:  +#+    +:+ +#++:++#+        +#+  +:+  
+#+  +#+   +#+     +#+ +#+        +#+    +#+ +#+    +#+ +#+    +#+ +#+             +#+#+#+#+#+
#+#   #+#  #+#     #+# #+#        #+#    #+# #+#    #+# #+#    #+# #+#                   #+#  
###    ### ###     ### ###        #########  ###    ###  ########  ###                   ###  

2023-10-12 10:23:36.222 [32m INFO[m [35m${sys:PID}[m [        main] [36mk.Kafdrop$EnvironmentSetupListener      [m : Initializing JAAS config
2023-10-12 10:23:36.233 [32m INFO[m [35m${sys:PID}[m [        main] [36mk.Kafdrop$EnvironmentSetupListener      [m : Env: null
2023-10-12 10:23:36.370 [32m INFO[m [35m1[m [kground-preinit] [36mo.h.v.i.u.Version                       [m : HV000001: Hibernate Validator 8.0.1.Final
2023-10-12 10:23:36.431 [32m INFO[m [35m1[m [           main] [36mo.s.b.StartupInfoLogger                 [m : Starting Kafdrop v4.0.0 using Java 17.0.8.1 with PID 1 (/kafdrop-4.0.0/kafdrop-4.0.0.jar started by root in /)
2023-10-12 10:23:36.437 [32m INFO[m [35m1[m [           main] [36mo.s.b.SpringApplication                 [m : No active profile set, falling back to 1 default profile: "default"
2023-10-12 10:23:38.393 [31mERROR[m [35m1[m [           main] [36mo.s.b.SpringApplication                 [m : Application run failed

java.lang.StackOverflowError
	at java.base/jdk.internal.ref.PhantomCleanable.<init>(PhantomCleanable.java:66)
	at java.base/jdk.internal.ref.CleanerImpl$PhantomCleanableRef.<init>(CleanerImpl.java:164)
	at java.base/java.lang.ref.Cleaner.register(Cleaner.java:220)
	at java.base/java.util.zip.ZipFile$ZipFileInflaterInputStream.<init>(ZipFile.java:435)
	at java.base/java.util.zip.ZipFile$ZipFileInflaterInputStream.<init>(ZipFile.java:428)
	at java.base/java.util.zip.ZipFile.getInputStream(ZipFile.java:395)
	at java.base/java.util.jar.JarFile.getInputStream(JarFile.java:844)
	at java.base/jdk.internal.loader.URLClassPath$JarLoader$2.getInputStream(URLClassPath.java:879)
        ....
```
and here a bit more detailed exception:
```
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:95)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class kotlin.reflect.jvm.internal.impl.builtins.jvm.JavaToKotlinClassMap
	at kotlin.reflect.jvm.internal.RuntimeTypeMapper.mapJvmClassToKotlinClassId(RuntimeTypeMapper.kt:274)
	at kotlin.reflect.jvm.internal.KClassImpl.getClassId(KClassImpl.kt:186)
	at kotlin.reflect.jvm.internal.KClassImpl.access$getClassId(KClassImpl.kt:44)
```

At the moment we reverted image to the version 3.31.0, however would be really good to know: shall we stick to this old version for a long period of time?

Thank you in advance!

--- a/README.md
+++ b/README.md
@@ -99,7 +99,7 @@ docker run -d --rm -p 9000:9000 \
     obsidiandynamics/kafdrop
 ```
 
-Lunch container with some specific JVM options:
+Launch container with some specific JVM options:
 ```sh
 docker run -d --rm -p 9000:9000 \
     -e KAFKA_BROKERCONNECT=<host:port,host:port> \

--- a/README.md
+++ b/README.md
@@ -95,16 +95,22 @@ Launch container in background:
 ```sh
 docker run -d --rm -p 9000:9000 \
     -e KAFKA_BROKERCONNECT=<host:port,host:port> \
-    -e JVM_OPTS="-Xms32M -Xmx64M" \
     -e SERVER_SERVLET_CONTEXTPATH="/" \
     obsidiandynamics/kafdrop
 ```
 
+Lunch container with some specific JVM options:
+```sh
+docker run -d --rm -p 9000:9000 \
+    -e KAFKA_BROKERCONNECT=<host:port,host:port> \
+    -e JVM_OPTS="-Xms32M -Xmx64M" \
+    -e SERVER_SERVLET_CONTEXTPATH="/" \
+    obsidiandynamics/kafdrop
+```
 Launch container in background with protobuff definitions:
 ```sh
 docker run -d --rm -v <path_to_protobuff_descriptor_files>:/var/protobuf_desc -p 9000:9000 \
     -e KAFKA_BROKERCONNECT=<host:port,host:port> \
-    -e JVM_OPTS="-Xms32M -Xmx64M" \
     -e SERVER_SERVLET_CONTEXTPATH="/" \
     -e CMD_ARGS="--message.format=PROTOBUF --protobufdesc.directory=/var/protobuf_desc" \
     obsidiandynamics/kafdrop
@@ -295,7 +301,7 @@ docker run -d --rm -p 9000:9000 \
 ##### Advanced configuration
 | Name                     |Description
 |--------------------------|-------------------------------
-| `JVM_OPTS`               |JVM options.
+| `JVM_OPTS`               |JVM options. E.g.```JVM_OPTS: "-Xms16M -Xmx64M -Xss360K -XX:-TieredCompilation -XX:+UseStringDeduplication -noverify"```
 | `JMX_PORT`               |Port to use for JMX. No default; if unspecified, JMX will not be exposed.
 | `HOST`                   |The hostname to report for the RMI registry (used for JMX). Defaults to `localhost`.
 | `KAFKA_PROPERTIES_FILE`  |Internal location where the Kafka properties file will be written to (if `KAFKA_PROPERTIES` is set). Defaults to `kafka.properties`.

--- a/docker-compose/kafka-kafdrop/docker-compose.yaml
+++ b/docker-compose/kafka-kafdrop/docker-compose.yaml
@@ -7,7 +7,6 @@ services:
       - "9000:9000"
     environment:
       KAFKA_BROKERCONNECT: "kafka:29092"
-      JVM_OPTS: "-Xms16M -Xmx48M -Xss180K -XX:-TieredCompilation -XX:+UseStringDeduplication -noverify"
     depends_on:
       - "kafka"
   kafka:

